src/topology/schema.rs

```diff
@@ -92,3 +98,177 @@ pub(super) fn merged_definition(inputs: &[OutputId], config: &topology::Config)
 
     definition
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+
+    use indexmap::IndexMap;
+    use serde::{Deserialize, Serialize};
+    use value::Kind;
+    use vector_core::{
+        config::{DataType, Input, Output},
+        source::Source,
+        transform::{Transform, TransformConfig, TransformContext},
+    };
+
+    use crate::config::{SourceConfig, SourceContext, SourceOuter, TransformOuter};
+
+    use super::*;
+
+    #[derive(Debug, Clone, Serialize, Deserialize)]
+    struct MockComponent {
+        #[serde(skip)]
+        outputs: Vec<Output>,
+    }
+
+    #[async_trait::async_trait]
+    #[typetag::serde(name = "mock")]
+    impl SourceConfig for MockComponent {
+        async fn build(&self, _: SourceContext) -> crate::Result<Source> {
+            unimplemented!()
+        }
+
+        fn outputs(&self) -> Vec<Output> {
+            self.outputs.clone()
+        }
+
+        fn source_type(&self) -> &'static str {
+            unimplemented!()
+        }
+    }
+
+    #[async_trait::async_trait]
+    #[typetag::serde(name = "mock")]
+    impl TransformConfig for MockComponent {
+        async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
+            unimplemented!()
+        }
+
+        fn outputs(&self, _: &Definition) -> Vec<Output> {
+            self.outputs.clone()
+        }
+
+        fn transform_type(&self) -> &'static str {
+            unimplemented!()
+        }
+
+        fn input(&self) -> Input {
+            unimplemented!()
+        }
+    }
+
+    #[test]
+    fn test_merged_definition() {
```

Comment:
This test doesn't cover all cases — yet. I intend on expanding it in follow-up PRs, but wanted to keep the ball moving forward for now by continuing on with follow-up PRs that add all relevant features needed to get a wholistic working schema system, before circling back and filling some testing holes.

Having that entire system working also allows for higher-level integration tests, which are easier to grasp.